src/containerapp/azext_containerapp/tests/latest/test_containerapp_connected_env_scenario.py

```diff
@@ -0,0 +1,369 @@
+            checks=[
+                JMESPathCheck('name', env_name),
+                JMESPathCheck('properties.provisioningState', "Succeeded"),
+                JMESPathCheck('extendedLocation.name', custom_location_id)
+            ])
+        
+        file_ref, dapr_file = tempfile.mkstemp(suffix=".yml")
+
+        dapr_yaml = """
+                name: statestore
+                componentType: state.azure.blobstorage
+                version: v1
+                metadata:
+                - name: accountName
+                  secretRef: storage-account-name
+                secrets:
+                - name: storage-account-name
+                  value: storage-account-name
+                """
+
+        daprloaded = yaml.safe_load(dapr_yaml)
+
+        with open(dapr_file, 'w') as outfile:
+            yaml.dump(daprloaded, outfile, default_flow_style=False)
+
+        containerapp_env = self.cmd('containerapp connected-env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+
+        while containerapp_env["properties"]["provisioningState"].lower() == "waiting":
+            time.sleep(5)
+            containerapp_env = self.cmd('containerapp connected-env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+
+        self.cmd('containerapp connected-env dapr-component set -n {} -g {} --dapr-component-name {} --yaml {}'
+                 .format(env_name, resource_group, dapr_comp_name, dapr_file.replace( os.sep, os.sep + os.sep)),
+                 checks=[
+                     JMESPathCheck('name', dapr_comp_name),
+                 ])
+
+        os.close(file_ref)
+
+        self.cmd('containerapp connected-env dapr-component list -n {} -g {}'.format(env_name, resource_group), checks=[
+            JMESPathCheck('length(@)', 1),
+            JMESPathCheck('[0].name', dapr_comp_name),
+        ])
+
+        self.cmd('containerapp connected-env dapr-component show -n {} -g {} --dapr-component-name {}'.format(env_name, resource_group, dapr_comp_name), checks=[
+            JMESPathCheck('name', dapr_comp_name),
+            JMESPathCheck('properties.version', 'v1'),
+            JMESPathCheck('properties.secrets[0].name', 'storage-account-name'),
+            JMESPathCheck('properties.metadata[0].name', 'accountName'),
+            JMESPathCheck('properties.metadata[0].secretRef', 'storage-account-name'),
+        ])
+
+        self.cmd('containerapp connected-env dapr-component remove -n {} -g {} --dapr-component-name {}'.format(env_name, resource_group, dapr_comp_name))
+
+        self.cmd('containerapp connected-env dapr-component list -n {} -g {}'.format(env_name, resource_group), checks=[
+            JMESPathCheck('length(@)', 0),
+        ])
+
+    @serial_test()
+    @live_only()  # generate_randomized_cert_name cause No match for the request (<Request (PUT) /my-connected-env/certificates/my-connected-e-clitest.rg0000-8d2d-6528?
```

Comment:
have to run with live_only()?

Reply:
Yes.
`generate_randomized_cert_name `cause No match for the request (<Request (PUT) /my-connected-env/certificates/my-connected-e-clitest.rg0000-8d2d-6528?
There is another test with certificate name not auto-generate, which can run in playback.